activitysim/benchmarking/componentwise.py

```diff
@@ -0,0 +1,601 @@
+
+    tracing.config_logger(basic=False)
+    config.filter_warnings()
+    logging.captureWarnings(capture=True)
+
+    # directories
+    for k in ["configs_dir", "settings_file_name", "data_dir", "output_dir"]:
+        logger.info("SETTING %s: %s" % (k, inject.get_injectable(k, None)))
+
+    log_settings = inject.get_injectable("log_settings", {})
+    for k in log_settings:
+        logger.info("SETTING %s: %s" % (k, config.setting(k)))
+
+    # OMP_NUM_THREADS: openmp
+    # OPENBLAS_NUM_THREADS: openblas
+    # MKL_NUM_THREADS: mkl
+    for env in ["MKL_NUM_THREADS", "OMP_NUM_THREADS", "OPENBLAS_NUM_THREADS"]:
+        logger.info(f"ENV {env}: {os.getenv(env)}")
+
+    np_info_keys = [
+        "atlas_blas_info",
+        "atlas_blas_threads_info",
+        "atlas_info",
+        "atlas_threads_info",
+        "blas_info",
+        "blas_mkl_info",
+        "blas_opt_info",
+        "lapack_info",
+        "lapack_mkl_info",
+        "lapack_opt_info",
+        "mkl_info",
+    ]
+
+    for cfg_key in np_info_keys:
+        info = np.__config__.get_info(cfg_key)
+        if info:
+            for info_key in ["libraries"]:
+                if info_key in info:
+                    logger.info(f"NUMPY {cfg_key} {info_key}: {info[info_key]}")
+
+    t0 = tracing.print_elapsed_time()
+
+    logger.info(f"MODELS: {config.setting('models')}")
+
+    if config.setting("multiprocess", False):
+        logger.info("run multi-process complete simulation")
+    else:
+        logger.info("run single process simulation")
+        pipeline.run(models=config.setting("models"))
+        pipeline.close_pipeline()
+
+    tracing.print_elapsed_time("prerun required models for checkpointing", t0)
+
+    return 0
+
+
+def run_multiprocess():
+    logger.info("run multiprocess simulation")
+    tracing.delete_trace_files()
+    tracing.delete_output_files("h5")
+    tracing.delete_output_files("csv")
+    tracing.delete_output_files("txt")
+    tracing.delete_output_files("yaml")
+    tracing.delete_output_files("prof")
+    tracing.delete_output_files("omx")
+
+    from activitysim.core import mp_tasks
+
+    injectables = {k: inject.get_injectable(k) for k in INJECTABLES}
+    mp_tasks.run_multiprocess(injectables)
+
+    assert not pipeline.is_open()
+
+    if config.setting("cleanup_pipeline_after_run", False):
+        pipeline.cleanup_pipeline()
+
+
+########
+
+
+def local_dir():
+    benchmarking_directory = workspace.get_dir()
+    if benchmarking_directory is not None:
+        return benchmarking_directory
+    return os.getcwd()
+
+
+def model_dir(*subdirs):
+    return os.path.join(local_dir(), "models", *subdirs)
+
+
+def template_setup_cache(
+    example_name,
+    component_names,
+    benchmark_settings,
+    benchmark_network_los,
+    config_dirs=("configs",),
+    data_dir="data",
+    output_dir="output",
+    settings_filename="settings.yaml",
+    skip_component_names=None,
+    config_overload_dir="dynamic_configs",
+):
+    """
+
```

Comment:
Comment needed on what this method does.  Considering this is one of two primary function calls from the benchmark templates, it would be helpful for any developers looking to make modifications.

Reply:
@jpn-- I think you missed this one